zephyr/CMakeLists.txt

```diff
@@ -482,13 +482,23 @@ zephyr_library_sources_ifdef(CONFIG_MATH_IIR_DF2T
 	${SOF_MATH_PATH}/iir_df2t_hifi3.c
 	${SOF_MATH_PATH}/iir_df2t.c
 )
-
-zephyr_library_sources_ifdef(CONFIG_COMP_ASRC
-	${SOF_AUDIO_PATH}/asrc/asrc.c
-	${SOF_AUDIO_PATH}/asrc/asrc_farrow_hifi3.c
-	${SOF_AUDIO_PATH}/asrc/asrc_farrow.c
-	${SOF_AUDIO_PATH}/asrc/asrc_farrow_generic.c
-)
+if(CONFIG_IPC_MAJOR_3)
+	zephyr_library_sources_ifdef(CONFIG_COMP_ASRC
+		${SOF_AUDIO_PATH}/asrc/asrc.c
+		${SOF_AUDIO_PATH}/asrc/asrc_ipc3.c
+		${SOF_AUDIO_PATH}/asrc/asrc_farrow_hifi3.c
+		${SOF_AUDIO_PATH}/asrc/asrc_farrow.c
+		${SOF_AUDIO_PATH}/asrc/asrc_farrow_generic.c
+	)
+elseif(CONFIG_IPC_MAJOR_4)
+	zephyr_library_sources_ifdef(CONFIG_COMP_ASRC
+		${SOF_AUDIO_PATH}/asrc/asrc.c
+		${SOF_AUDIO_PATH}/asrc/asrc_ipc4.c
+		${SOF_AUDIO_PATH}/asrc/asrc_farrow_hifi3.c
+		${SOF_AUDIO_PATH}/asrc/asrc_farrow.c
+		${SOF_AUDIO_PATH}/asrc/asrc_farrow_generic.c
```

Comment:
Copy/paste, seriously? Move the common .c files out of the CONFIG_IPC_.

Don't nest `CONFIG_COMP_ASRC` inside `CONFIG_IPC_MAJOR_x`, do it the other way round.

Reply:
or should I move it to line 718 like you did in https://github.com/thesofproject/sof/pull/8425?
`if(CONFIG_IPC_MAJOR_3)
	zephyr_library_sources_ifdef(CONFIG_SAMPLE_SMART_AMP
		${SOF_SAMPLES_PATH}/audio/smart_amp_test_ipc3.c
	)

	zephyr_library_sources_ifdef(CONFIG_COMP_TDFB
	${SOF_AUDIO_PATH}/tdfb/tdfb_ipc3.c
	)
elseif(CONFIG_IPC_MAJOR_4)
	zephyr_library_sources_ifdef(CONFIG_SAMPLE_SMART_AMP
		${SOF_SAMPLES_PATH}/audio/smart_amp_test_ipc4.c
	)

	zephyr_library_sources_ifdef(CONFIG_COMP_TDFB
	${SOF_AUDIO_PATH}/tdfb/tdfb_ipc4.c
	)
endif()`
or line 452?
there are several styles in this document.  the volume & src & module_adapter is copy/past, and tdfb & smart_amp_test & eq is another format, and if I nest CONFIG_IPC_MAJOR_x inside CONFIG_COMP_ASRC then it would be third style

Reply:
It does matter to me a great lot because tracking all these configuration inconsistencies is what makes the cleanup very time-consuming. It would be very quick and easy otherwise.